### DISCUSSION TITLE
Decompressing data results in an invalid blob

### DISCUSSION THREAD

**Discussion by CarloPoletto** [101arrowz/fflate#119] [Q&A] [answered]
I am developing a web application which has to exchange large files between back-end and front-end. For this reason, on the back-end I have developed an algorithm in python to create a zip file with a single file inside it (no folders). Then, on the front-end side, I download this file and I tried to decompress it in two different ways: using JSZip and fflate. With JSZip the decompression happens correctly but it is rather slow while with fflate the decompression happens very quickly without errors but this generates a small and wrong Blob object. Here are the two javascript algorithms I use for decompression:

```
let blob: Blob = ... //compressed data
let jsZip = await new JSZip().loadAsync(blob);
let file = Object.values(jsZip.files)[0];
let out = await file.async("blob"); //decompressed data
console.log(out.size); //return 308219807
```

```
let blob: Blob = ... //compressed data
let arrayBuffer = await blob.arrayBuffer();
let uint8Array = new Uint8Array(arrayBuffer);
let outArray = fflate.decompressSync(uint8Array);
let out = new Blob([outArray ]); //decompressed data
console.log(out.size); //return 1501
```

Am I wrong in using the library?

**Comment by 101arrowz** [comment, marked as answer]
This is a similar issue to #111. You can't use `decompressSync` for ZIP files, you need to use `unzip` or `unzipSync`. Here's some code that should work:

```js
import { unzip } from 'fflate';

const uint8Array = new Uint8Array(await blob.arrayBuffer());
const unzipped = await new Promise((resolve, reject) => {
  unzip(uint8Array, (err, result) => err ? reject(err) : resolve(result));
});
// The above code decompresses in another thread, which prevents UI freezing
// If your files are relatively small, this may be faster:
// const unzipped = unzipSync(uint8Array);

const fileBuffer = Object.values(unzipped)[0];
let out = new Blob([fileBuffer]);
console.log(out.size);
```
